torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -231,6 +231,15 @@ c10::optional<at::Tensor> runTorchBackendForOnnx(
       return c10::optional<at::Tensor>(updated_val);
     }
     return c10::nullopt;
+  } else if (node->kind() == onnx::Reshape) {
+    assert(inputTensorValues.size() == 2);
+    updated_val = inputTensorValues[0];
+    std::vector<int64_t> shape(inputTensorValues[1].sizes()[0], 0);
```

Comment:
Consider adding a check if the number of elements between old and new shape match.

Reply:
The new shape array is defined to have the same size as the initial shape input
std::vector<int64_t> shape(inputTensorValues[1].sizes()[0], 0);
And when filling the shape array, we iterate over the initial shape input
for (size_t i = 0; i < inputTensorValues[1].sizes()[0]; ++i)

Reply:
That's fine. But consider a situation where a bad input shape comes in which has number of elements different than the original input array. That could crash the program in the loop below, most probably. I was suggesting having a check that would validate that the two inputs agree on the shape so that we can error out gracefully if they don't.